task3_2/js/function.js

```diff
@@ -0,0 +1,227 @@
+//isArray function
+function isArrayPrototype(param) {
+  return Object.prototype.toString.call(param) === '[object Array]';
+}
+
+function isArrayInsanceof(param) {
+  return param instanceof Array;
+}
+
+function isArrayConstructor(param) {
+  return Boolean(param) && param.constructor === Array; //
+}
+
+function testIsArray(func) {
+  var a = [1,2,3]
+  var b = 3;
+  var c = { name: "name" };
+  var d = [];
+  var str = "sdfsfd";
+  console.log("test iaArray:");
+  console.log(func(a));
+  console.log(func(d));
+  console.log(func(b));
+  console.log(func(c));
+  console.log(func(str));
+}
+
+//range function
+function range(a, b, c) {
```

Comment:
so many cases...
for loop will do everything for you
range(10, 1) returns []
range(-10, -1) returns [-10 ... -2]

function is too complicated